tests/ui/rfcs/rfc-0000-never_patterns/unreachable.rs

```diff
@@ -0,0 +1,31 @@
+// revisions: normal exh_pats
+//[normal] check-pass
+#![feature(never_patterns)]
+#![allow(incomplete_features)]
+#![cfg_attr(exh_pats, feature(exhaustive_patterns))]
+#![allow(dead_code, unreachable_code)]
+#![deny(unreachable_patterns)]
+
+#[derive(Copy, Clone)]
+enum Void {}
+
+fn main() {
+    let res_void: Result<bool, Void> = Ok(true);
+
+    match res_void {
+        Ok(_x) => {}
+        Err(!),
```

Comment:
Ok, just double checking my understanding -- with exhaustive patterns, we don't need to care about never patterns b/c the scrutinee already disqualifies those patterns?

Reply:
Yes. Said differently, never patterns are shorthand syntax: `Err(!)` is short for `Err(_) => unreachable_unchecked!()`. The rules for which arms are required are orthogonal, and here `exhaustive_patterns` says "we don't need that arm". I'll go update the tracking issue to clarify that while I'm at it